test/system/bans_test.rb

```diff
@@ -117,8 +135,10 @@ def test_banned_user_can_see_the_ban_notification_for_longest_repetitive_ban
       invoices until 2010-07-10.
     TEXT
 
+
     visit auctions_path
     assert(page.has_css?('div.ban', text: text))
+    has_no_css?('div.ban > p')
```

Comment:
The intention is not clear. Consider testing it from a user viewpoint.

Reply:
Precisely from user's viewpoint we are testing here
1) user can see the text message on total ban
2) message on violations count didn't displayed.

So I'd rather keep it that way.

Reply:
A user doesn't see HTML code, still :). This test is fragile, since it relies on HTML structure too heavily.

> message on violations count didn't displayed.

Can we document this in code?

Like `assert_no_text 'message on violations'`?